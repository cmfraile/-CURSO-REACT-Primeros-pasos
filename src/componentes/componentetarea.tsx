export const FirstApp = () => <h1>First app</h1>