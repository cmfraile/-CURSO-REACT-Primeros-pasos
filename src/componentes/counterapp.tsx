interface counterif {contador:number};
export const counter = ({contador}:counterif) => {

    return(
        <>
            <h1>Counterapp</h1>
            <h2>{contador}</h2>
        </>
    )
    
}