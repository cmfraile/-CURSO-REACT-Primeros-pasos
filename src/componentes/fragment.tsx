import { random } from "underscore";
import PropTypes from 'prop-types';

export const fragmentest = () => 
<>
    <h1>Esta mierda chucha</h1>
    <p>chucha gushi</p>
</>

interface tsfragmentif {title?:string,dado2?:number}
export const tsfragment = ({title='undefined',dado2}:tsfragmentif) => {

    console.log(title,dado2);

    const var1 = {
        nombre: title,
        rango: 2,
        dado : random(1,10),
        dado2 : dado2
    }
    
    return (
        <>
        <h1>fragmento</h1>
        <p>Soy el supersayayin { var1.nombre }</p>
        <code>{ JSON.stringify(var1) }</code>
        </>
    )
}