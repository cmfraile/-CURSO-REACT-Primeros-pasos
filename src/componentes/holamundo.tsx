export const holamundoApp = () =>  {return(<h1>Hola mundo</h1>)}
export const holamundoApp2 = () =>  <h1>Hola mundo</h1>
