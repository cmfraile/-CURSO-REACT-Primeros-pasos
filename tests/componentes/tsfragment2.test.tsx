import { render , screen } from '@testing-library/react';
import { tsfragment as Tsf } from '../../src/componentes/fragment';
import React from 'react'

const main = () => {

    const title = 'Soy el supersayayin Goku'

    test('debe de hacer match con el snapshot',() => {
        expect(render(<Tsf title={'Goku'}/>).container).toMatchSnapshot();
    });

    test(`debe de mostrar el titulo [${title}]`,() => {
        render(<Tsf title={'Goku'}/>);
        expect(screen.getByText(title) ).toBeTruthy();
    });

    test('debe de situar el titulo en el <p>',() => {
        render(<Tsf title={'Goku'}/>);
        expect(screen.getByTestId('testitle').innerHTML).toContain(title);

        //Si estuviera en el H1:
        //expect(screen.getByRole('heading',{level:1}).innerHTML).toContain(title);
    });

    test('Debe de venir el titulo por props',() => {
        render(<Tsf title={'Goku'}/>);
        expect(screen.getAllByText(title).length).toBe(1);
    })

}

describe('Pruebas de <TSF>',main);