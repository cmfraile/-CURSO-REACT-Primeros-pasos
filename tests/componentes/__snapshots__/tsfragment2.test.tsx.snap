// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <TSF> debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    fragmento
  </h1>
  <p
    data-testid="testitle"
  >
    Soy el supersayayin 
    Goku
  </p>
  <code>
    {"nombre":"Goku","rango":2}
  </code>
</div>
`;
